x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/QuerySettings.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical;
+
+import org.elasticsearch.xpack.esql.core.expression.Alias;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+
+import java.util.List;
+import java.util.Objects;
+
+public class QuerySettings {
```

Comment:
I'm not treating this as a command or as part of the AST in general. 
It's on purpose, we don't want it to part of the query (for now at least), so it doesn't need resolution/serialization/traversal for now.

Ah, I'm not calling it `Set`. There is only one `Set` in Java.
Suggestions for a better name are welcome of course